test/deployment.js

```diff
@@ -19,26 +19,18 @@ describe("deployment", function () {
     before(function (done) {
         this.timeout(30 * 1000);
 
-        api.deployment.deploy(gitUrl1, function (err) {
-            if (err) {
-                return done(err);
-            }
-
-            api.deployment.deploy(gitUrl2, function (err) {
-                if (err) {
-                    return done(err);
-                }
-
-                api.deployment.list(function (err, result) {
-                    if (err) {
-                        return done(err);
-                    }
-
-                    deploymentList = result.data;
-                    done();
-                });
-            });
-        });
+        api.deployment.deployAsync(gitUrl1)
+            .then(function () {
+                return api.deployment.deployAsync(gitUrl2);
+            })
+            .then(function () {
+                return api.deployment.listAsync();
+            })
+            .then(function (result) {
+                deploymentList = result.data;
+                done();
+            })
+            .catch(done);
```

Comment:
Arrow functions would improve this further, but that would mean a transpiler or dropping old Node versions. It's still a big reduction in nesting though!